test/functional/options/fillchars_spec.lua

```diff
@@ -0,0 +1,39 @@
+local helpers = require('test.functional.helpers')(after_each)
+local Screen = require('test.functional.ui.screen')
+local clear, execute = helpers.clear, helpers.execute
+
+if helpers.pending_win32(pending) then return end
+
+describe("'fillchars'", function()
+  local screen
+
+  before_each(function()
+    clear()
+    screen = Screen.new(25, 5)
+    screen:attach()
+  end)
+
+  after_each(function()
+    screen:detach()
+  end)
+
+  describe('the "eob" option', function()
+    it('renders empty lines at the end of the buffer with eob', function()
+      screen:expect([[
+        ^                         |
+        ~                        |
+        ~                        |
+        ~                        |
+                                 |
+      ]])
+      execute('set fillchars+=eob:\\ ')
```

Comment:
Ah yes sorry, I mistakenly tried this with multiple characters. Have added a multibyte character test case now.